### ISSUE TITLE
Insert will break with a different count of keys

### ISSUE THREAD

**Issue by jamesser** [dmfay/massive-js#467]
I was talking with @markgeraty about #465 earlier today, and checked out the fix (#466)

I think there will still be an issue if you insert a different number of parameters for each element of the array.

For example,

```javascript
// For a table consisting of both `a` and `b` columns, assuming `b` can be null
// this.records = [{ a: 'apple' }, { b: 'bear', a: 'aardvark' }];

const keys = _.keys(this.records[0]);

// keys = ['a']

this.columns = _.map(keys, key => `"${key}"`);
this.params = _.reduce(this.records, (params, element) => params.concat(keys.map(key => element[key])), []);

// this.params = ['apple', 'aardvark']
```

Here, the values for column b in our second record have been dropped resulting in our table 

a | b
--- | ---
apple | 
aardvark |

**Comment by dmfay** [comment]
That actually isn't supposed to work! As you've noticed it leads to inconsistent behavior, which is why if you try to `INSERT INTO mytable (id, name) VALUES (1, 'alice'), (2)` in psql, Postgres will abort with an error telling you the `VALUES` lists must all be the same length. You'd have gotten that error from Massive before too, since it just concatenated the object values arrays. With the fix for #465 it's looking up the keys instead and generating `undefined` for any missing, which satisfies the length test for Postgres but will happily insert your malformed data so long as it doesn't violate not null constraints.

It seems like it should be possible to support merging keysets so cases like `db.mytable.insert({field1: 'one'}, {field2: 'two'})` would generate `('one', null), (null, 'two')` `VALUES` lists. I'll have to think about that one a bit.

**Comment by jamesser** [comment]
Sounds good.  Maybe just a documentation change?  I was thrown by this [paragraph](https://dmfay.github.io/massive-js/persistence.html)

> insert writes an object into your table. The object’s keys should match your column names exactly. You do not have to perform a complete insert and specify every single field in your object; as long as you ensure that no NOT NULL constraints will be violated, you can include as many or as few fields as you need. When you insert an object, the record will be returned as an object. If you have an autogenerated or serial primary key or default fields, they will have been calculated and added.
